bson/bsoncodec/uint_codec.go

```diff
@@ -24,7 +24,7 @@ type UIntCodec struct {
 var (
 	defaultUIntCodec = NewUIntCodec()
 
-	_ ValueCodec  = defaultUIntCodec
+	_ valueCodec  = defaultUIntCodec
```

Comment:
Could we add a comment describing the compile check?

Reply:
This compile-time assertion is actually redundant because the codec is already passed into `RegisterTypeEncoder` and `RegisterTypeDecoder`, so I will remove it. However, I will add a comment to the `typeDecoder` assertion below.